src/main/java/com/scrive/pdftools/tools/LambdaInterface.java

```diff
@@ -0,0 +1,137 @@
+    private static final String RESP_STATUS_CODE = "statusCode";
+    private static final String RESP_IS_B64_ENCODED = "isBase64Encoded";
+
+    private static void handleFlattenAction(JSONObject callJSON, OutputStream outputStream) {
+        getS3FileName(callJSON, outputStream).ifPresent(s3fileName -> {
+            getFileContent(s3fileName, outputStream).ifPresent(fileContent -> {
+                final JSONObject jsonObject;
+                try {
+                    jsonObject = new JSONObject(fileContent);
+                } catch (Exception e) {
+                    e.printStackTrace();
+                    respondWithError(outputStream, "Can't parse spec as JSON");
+                    return;
+                }
+
+                try {
+                    System.out.println("DEBUG: Started flattening action on document");
+                    byte[] out1 = FlattenerHandler.execute(jsonObject);
+                    String resultFileName = RESP_FILE_NAME_PREFIX + s3fileName;
+                    S3ForLambda.putStringToAmazonFile(resultFileName, out1);
+                    JSONObject response = new JSONObject();
+                    response.put(RESP_RESULT_S3_FILE_NAME, resultFileName);
+                    response.put(RESP_SUCCESS, true);
+                    respondWith(outputStream, response);
+                    System.out.println("DEBUG: Finished flattening action on document");
+                } catch (Exception e) {
+                    e.printStackTrace();
+                    respondWithError(outputStream, e.getMessage());
+                }
+            });
+        });
+    }
+
+    public void handleRequest(InputStream inputStream, OutputStream outputStream, Context context) {
+
+        JSONObject callJSON;
+        try {
+            String fs = IOUtils.toString(inputStream, StandardCharsets.UTF_8);
+            JSONObject inputJSON = new JSONObject(fs);
+            boolean hasRequestBodyHiddenInJSON = inputJSON.has(RQ_PARAM_ACTION_BODY);
+            if (hasRequestBodyHiddenInJSON) {
+                callJSON = new JSONObject(inputJSON.getString(RQ_PARAM_ACTION_BODY));
+            } else {
+                callJSON = inputJSON;
+            }
+        } catch (Exception e) {
+            e.printStackTrace();
+            respondWithError(outputStream, "Invalid input - call spec has to be a json");
+            return;
+        }
+
+        if (!callJSON.has(RQ_PARAM_ACTION)) {
+            respondWithError(outputStream, "Invalid action. Action has to be set");
+            return;
+        }
+
+        switch (callJSON.getString(RQ_PARAM_ACTION)) {
+            case RQ_PARAM_ACTION_FLATTEN:
+                System.out.println("DEBUG: Processing flatten action");
+                handleFlattenAction(callJSON, outputStream);
+                return;
+            default:
+                System.out.println(String.format("ERROR: Action %s isn't permitted", callJSON.getString(RQ_PARAM_ACTION)));
```

Comment:
maybe a more standard way would be ending with break sine we don't have any specific value to return
```suggestion
                break;
            default:
                System.out.println(String.format("ERROR: Action %s isn't permitted", callJSON.getString(RQ_PARAM_ACTION)));
                break;
```

Reply:
Yes. I changed to using the more proper break;